main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintSolver.scala

```diff
@@ -277,7 +277,7 @@ object ConstraintSolver {
       // Apply substitution now
       val tpes = tpes0.map(subst0.apply)
       // Ensure that simplification for constructor parameters is done
-      val allKnown = tpes.forall(isKnown)
+      val allKnown = tpes.forall(t => isKnown(t) && t.typeVars.forall(isKnown))
```

Comment:
What's the contract of the `isKnown` function? It seems like we ought to change it so it implements essentially what you have here.

Reply:
isKnown is a higher-level constraint that allows variables to be substituted as long as the constraint solver is running, so that we can go back later, stronger, to this solving. Since we apply substitution beforehand, isKnown was returning false if the type remained a variable.